### PR TITLE
Add Treasurer to State

### PR DESCRIPTION
**Pull request by randomshinichi**
Implements VPROT-124.
An "xtra" tree is added to the State,  where misc things like the Treasurer address will be stored.

### CODE REVIEW

**Comment by jordipainan** [comment]
_self-reminder: Squash the commits_

**Review by jordipainan — APPROVED** [review]

File: vochain/app.go [hunk 1]
```diff
@@ -350,6 +350,15 @@ func (app *BaseApplication) InitChain(req abcitypes.RequestInitChain) abcitypes.
 		}
 	}
 
+	// set treasurer address
+	if !ethcommon.IsHexAddress(genesisAppState.Treasurer) {
+		log.Fatalf("Treasurer should be a valid Ethereum address: %s", err)
```
Comment (p4u) [inline vochain/app.go]: log messages should not start with capital letter

File: vochain/app.go [hunk 2]
```diff
@@ -350,6 +350,15 @@ func (app *BaseApplication) InitChain(req abcitypes.RequestInitChain) abcitypes.
...
+	}
+	log.Infof("adding genesis treasurer %s", genesisAppState.Treasurer)
+	if err := app.State.SetTreasurer(ethcommon.HexToAddress(genesisAppState.Treasurer)); err != nil {
+		log.Fatalf("Could not set State.Treasurer from genesis file: %s", err)
```
Comment (p4u) [inline vochain/app.go]: same

File: vochain/state.go [hunk 3]
```diff
@@ -481,6 +501,40 @@ func (v *State) IsOracle(addr common.Address) (bool, error) {
 	}(), nil
 }
 
+// SetTreasurer saves the Treasurer address to the state
+func (v *State) SetTreasurer(address common.Address) error {
```
Comment (p4u) [inline vochain/state.go]: Do we really need to export this method?

**Review by p4u — APPROVED** [review]

File: vochain/state.go (hunk 3 above)
Comment (randomshinichi) [reply vochain/state.go]: True that.

**Pull request merged**